app/controllers/filters_controller.rb

```diff
@@ -1,37 +1,64 @@
 class FiltersController < ApplicationController
-  before_action :load_data, only: %i(search search_by_price_range)
+  before_action :load_data, :search_params, only: :search
 
   def search
-    case params[:search_type]
-    when Settings.search_type.name
-      search_by_field Settings.search_type.name
-    when Settings.search_type.brand
-      search_by_field Settings.search_type.brand
-    when Settings.search_type.category
-      search_by_field Settings.search_type.category
-    end
+    @products = filter_products
+    @pagy, @products = pagy(@products, items: Settings.pagy.product.per_page)
     render "products/index"
   end
 
-  def search_by_price_range
-    start_price = params[:start].to_i
-    end_price = params[:end].to_i
-    @pagy, @products = pagy(Product.find_by_price_range(start_price, end_price)
-                                   .order_by_name,
-                            items: Settings.pagy.product.per_page)
-    render "products/index"
+  private
+
+  def search_params
+    @name = params[:name]
+    @brand = params[:brand]
+    @category = params[:category]
+    @start_price = params[:start_price].to_i
+    @end_price = params[:end_price].to_i
+    @increment = params[:increment]
   end
 
-  private
+  def filter_products
+    products = Product.all
+    products = filter_by_name products
```

Comment:
Ai lại làm như này em. Chuyển hết về scope r viết chain scope đi. Hoặc bên trên đã load ra toàn bộ sản phẩm thì bên dưới thao tác với mảng. Xử lý như này ko tối ưu